ta/remoteproc/remoteproc_core.c

```diff
@@ -540,6 +544,64 @@ static TEE_Result remoteproc_verify_firmware(struct remoteproc_context *ctx,
 	return res;
 }
 
+static paddr_t remoteproc_da_to_pa(uint32_t da, size_t size,
+				   struct remoteproc_context *ctx)
+{
+	uint32_t param_types = TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_INPUT,
+					       TEE_PARAM_TYPE_VALUE_INPUT,
+					       TEE_PARAM_TYPE_VALUE_INPUT,
+					       TEE_PARAM_TYPE_VALUE_OUTPUT);
+	TEE_Param params[TEE_NUM_PARAMS] = { };
+	TEE_Result res = TEE_ERROR_GENERIC;
+
+	/*
+	 * The ELF file contains remote processor device addresses, that refer
+	 * to the remote processor memory space.
+	 * A translation is needed to get the corresponding physical address.
+	 */
+
+	params[0].value.a = ctx->fw_id;
+	params[1].value.a = da;
+	params[2].value.a = size;
+
+	res = TEE_InvokeTACommand(pta_session, TEE_TIMEOUT_INFINITE,
+				  PTA_RPROC_FIRMWARE_DA_TO_PA,
+				  param_types, params, NULL);
+	if (res != TEE_SUCCESS) {
+		EMSG("Failed to translate device address %#"PRIx32, da);
+		return 0;
```

Comment:
maybe add an inline comment here: `/* By convention in core, physical address 0 is invalid */`.

Reply:
done